src/app/(full-page)/genres/page.tsx

```diff
@@ -1,7 +1,37 @@
-export default function GenresPage() {
+import { GetServerSideProps } from "next";
+import { knownGenres, Genre } from "../../../lib/utils";
+import { GenreButton } from "../../../components/genre-button";
+import { useState } from "react";
+
+interface GenresPageProps {
+  genres: Genre[];
+}
+
+const GenresPage = ({ genres }: GenresPageProps) => {
+  const [selectedGenre, setSelectedGenre] = useState<Genre | null>(null);
+
   return (
-    <div>
-      <span>Genres</span>
+    <div className="flex flex-wrap justify-center gap-4 p-4">
+      {genres.map((genre) => (
+        <GenreButton
+          key={genre}
+          genre={genre}
+          selected={selectedGenre === genre}
+          onClick={() => setSelectedGenre(genre)}
+        />
+      ))}
     </div>
   );
-}
+};
+
+export const getServerSideProps: GetServerSideProps = async () => {
```

Comment:
Same as in the movies page, pls don't use getServerSideProps
use a server component instead